uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -1786,5 +1899,257 @@ public function testGetTotalTOILBalanceChangeForContactWithinAGivenPeriodAndWith
     );
     $this->assertEquals(3, $totalBalanceChange);
   }
+
+  public function testRecalculateExpiredBalanceChangesForLeaveRequestPastDates() {
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('-50 days'),
+      'end_date' => CRM_Utils_Date::processDate('-20 days')
+    ]);
+
+    $periodEntitlement1 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => 1,
+      'period_id' => $absencePeriod->id,
+      'type_id' => 1,
+    ]);
+
+    $periodEntitlement2 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => 1,
+      'period_id' => $absencePeriod->id,
+      'type_id' => 2,
+    ]);
+
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+
+    $balanceChange1 = $this->createExpiredBroughtForwardBalanceChange(
+      $periodEntitlement1->id,
+      1,
+      1,
+      30
+    );
+
+    //This balance change expired before the leave request dates so it will not be affected
+    $balanceChange2 = $this->createExpiredBroughtForwardBalanceChange(
```

Comment:
please, to improve readability, rename the balance change variables to things like 'broughtForwardPeriod1' or 'accreudTroilPeriod1'.